core/api/transport/impl/listener_impl.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/transport/impl/listener_impl.hpp"
+
+#include "api/transport/impl/session_impl.hpp"
+
+namespace kagome::server {
+  using kagome::server::SessionImpl;
+
+  ListenerImpl::ListenerImpl(ListenerImpl::Context &context,
+                             const ListenerImpl::Endpoint &endpoint, Configuration config)
+      : context_(context), acceptor_(context_, endpoint), config_(config) {
+    onError().connect([this](outcome::result<void> err) {
+      // TODO(yuraz): pre-230 log error
+      stop();
+    });
+  }
+
+  void ListenerImpl::doAccept() {
+    acceptor_.async_accept(
+        [this](boost::system::error_code ec, Session::Socket socket) {
+          if (!ec) {
```

Comment:
change to 
```suggestion
          if (ec) {
             return onError()(outcome::failure(ec));
          }
```
With such approach you will reduce nesting

Reply:
double comment